### ISSUE TITLE
[FEATURE] Click once instead of twice

### ISSUE THREAD

**Issue by algj** [KrLite/Tap-Tab#11]
### Feature Description

Allow clicking once instead of twice to rotate hotbar.

### Propose Reason

Self explanatory - if you can configure other key than TAB, some people may set it to some keybind specifically for switching hotbar, it would be useless for them to click it twice.

**Comment by KrLite** [comment]
Good point. But the problem is that if this feature is adopted, we will need a config screen to somehow let the player choose if double-tap is enabled, which I think too verbose for a single switch entry. However, I'll consider that.

**Comment by JasdanVM** [comment]
> Good point. But the problem is that if this feature is adopted, we will need a config screen to somehow let the player choose if double-tap is enabled, which I think too verbose for a single switch entry. However, I'll consider that.

No matter how many options are presented to the user, having more options is always nice, at least in my opinion it's better to have the option and not use it, than being uinable to customize it that way.

**Comment by KrLite** [comment]
> > Good point. But the problem is that if this feature is adopted, we will need a config screen to somehow let the player choose if double-tap is enabled, which I think too verbose for a single switch entry. However, I'll consider that.
> 
> 
> 
> No matter how many options are presented to the user, having more options is always nice, at least in my opinion it's better to have the option and not use it, than being uinable to customize it that way.

I agree, but adding an option screen comes with a lot of code and maintenance pressure, so I always prefer the simplest solution. Maybe we can come up with 2 or 3 more toggles to add to the config to convince me to have a screen.

**Comment by JasdanVM** [comment]
> > > Good point. But the problem is that if this feature is adopted, we will need a config screen to somehow let the player choose if double-tap is enabled, which I think too verbose for a single switch entry. However, I'll consider that.
> > 
> > 
> > No matter how many options are presented to the user, having more options is always nice, at least in my opinion it's better to have the option and not use it, than being uinable to customize it that way.
> 
> I agree, but adding an option screen comes with a lot of code and maintenance pressure, so I always prefer the simplest solution. Maybe we can come up with 2 or 3 more toggles to add to the config to convince me to have a screen.

I think I have a neat solution, even more Keybinds, in relation to my post regarding alternative scrolling methods, I suggested having 5 keybinds, well, I think having 7 would actually be a great alternative to a config screen, let me explain:
 
1,2,3: The first 3 Keybinds would remain as they currently work in the mod, respecting your original vision.

4,5,6,7: This would be the additional, brand new keybinds assigned by default to the Arrow Keys, this would only need to be clicked once, _Arrow's Up and Down could cycle the entire hotbar, while Arrows Left and Right cycle the main held item_ and viceversa.

**Comment by KrLite** [comment]
> > > > Good point. But the problem is that if this feature is adopted, we will need a config screen to somehow let the player choose if double-tap is enabled, which I think too verbose for a single switch entry. However, I'll consider that.
> 
> > > 
> 
> > > 
> 
> > > No matter how many options are presented to the user, having more options is always nice, at least in my opinion it's better to have the option and not use it, than being uinable to customize it that way.
> 
> > 
> 
> > I agree, but adding an option screen comes with a lot of code and maintenance pressure, so I always prefer the simplest solution. Maybe we can come up with 2 or 3 more toggles to add to the config to convince me to have a screen.
> 
> 
> 
> I think I have a neat solution, even more Keybinds, in relation to my post regarding alternative scrolling methods, I suggested having 5 keybinds, well, I think having 7 would actually be a great alternative to a config screen, let me explain:
> 
>  
> 
> 1,2,3: The first 3 Keybinds would remain as they currently work in the mod, respecting your original vision.
> 
> 
> 
> 4,5,6,7: This would be the additional, brand new keybinds assigned by default to the Arrow Keys, this would only need to be clicked once, _Arrow's Up and Down could cycle the entire hotbar, while Arrows Left and Right cycle the main held item_ and viceversa.

That's practical.

**Comment by KrLite** [comment]
Or we can provide two versions of each keybind, which respectively respond to single and double tap events. Single taps can default to arrow keys, and double taps defaults to tab (+ modifier). And I'm thinking of combining modifiers with the key so that we can preview keybind conflicts and decompose keybind configurations.
